src/Components/WebAssembly/Build/test/BuildIntegrationTests/PublishIntegrationTest.cs

```diff
@@ -433,6 +432,109 @@ public async Task Publish_HostedApp_WithNoBuild_Works()
+            var resxfileInProject = Path.Combine(standaloneProjectDirectory , "Resources.ja.resx.txt");
+            File.Move(resxfileInProject, Path.Combine(standaloneProjectDirectory, "Resource.ja.resx"));
+
+            var standaloneProjFile = Path.Combine(standaloneProjectDirectory, "standalone.csproj");
+            var existing = File.ReadAllText(standaloneProjFile);
+            var updatedContent = @"
+<PropertyGroup>
+    <DefineConstants>$(DefineConstants);REFERENCE_classlibrarywithsatelliteassemblies</DefineConstants>
+</PropertyGroup>
+<ItemGroup>
+    <ProjectReference Include=""..\classlibrarywithsatelliteassemblies\classlibrarywithsatelliteassemblies.csproj"" />
+</ItemGroup>";
+            var updated = existing.Replace("<!-- Test Placeholder -->", updatedContent);
+            File.WriteAllText(standaloneProjFile, updated);
+
+            var result = await MSBuildProcessManager.DotnetMSBuild(project, "Build", $"/restore");
+
+            Assert.BuildPassed(result);
+
+            result = await MSBuildProcessManager.DotnetMSBuild(project, "Publish", "/p:BuildProjectReferences=false");
+
+            var publishDirectory = project.PublishOutputDirectory;
+            // Make sure the main project exists
+            Assert.FileExists(result, publishDirectory, "blazorhosted.dll");
+
+            var blazorPublishDirectory = Path.Combine(publishDirectory, "wwwroot");
+            Assert.FileExists(result, blazorPublishDirectory, "_framework", "blazor.boot.json");
+            Assert.FileExists(result, blazorPublishDirectory, "_framework", "_bin", "ja", "standalone.resources.dll");
+            Assert.FileExists(result, blazorPublishDirectory, "_framework", "_bin", "fr", "Microsoft.CodeAnalysis.resources.dll");
+
+            var bootJson = Path.Combine(project.DirectoryPath, blazorPublishDirectory, "_framework", "blazor.boot.json");
+            var bootJsonFile = JsonSerializer.Deserialize<GenerateBlazorBootJson.BootJsonData>(File.ReadAllText(bootJson), new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            var satelliteResources = bootJsonFile.resources.satelliteResources;
+
+            Assert.Contains("es-ES", satelliteResources.Keys);
+            Assert.Contains("es-ES/classlibrarywithsatelliteassemblies.resources.dll", satelliteResources["es-ES"].Keys);
+            Assert.Contains("fr", satelliteResources.Keys);
+            Assert.Contains("fr/Microsoft.CodeAnalysis.CSharp.resources.dll", satelliteResources["fr"].Keys);
+            Assert.Contains("ja", satelliteResources.Keys);
+            Assert.Contains("ja/standalone.resources.dll", satelliteResources["ja"].Keys);
+
+            VerifyServiceWorkerFiles(result, blazorPublishDirectory,
+                serviceWorkerPath: Path.Combine("serviceworkers", "my-service-worker.js"),
+                serviceWorkerContent: "// This is the production service worker",
+                assetsManifestPath: "custom-service-worker-assets.js");
```

Comment:
How come this template produces a service worker? Looks like it’s just the hosted template, not a PWA.

Reply:
The standalone app has a service worker, no? https://github.com/dotnet/aspnetcore/blob/blazor-wasm/src/Components/WebAssembly/Build/testassets/standalone/standalone.csproj#L7